plugins/modules/vm_portgroup_info.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2023, Ansible Cloud Team (@ansible-collections)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: vm_portgroup_info
+version_added: '1.4.0'
+short_description: Returns information about the portgroups of virtual machines
+description:
+    - Returns information about the standard or distributed portgroups of virtual machines.
+author:
+    - Ansible Cloud Team (@ansible-collections)
+requirements:
+    - vSphere Automation SDK
+options:
+    vm_names:
+        description:
+            - VM names for retrieving the information about portgroup
+        required: true
+        type: list
+        elements: str
+extends_documentation_fragment:
+    - vmware.vmware.vmware_rest_client.documentation
+'''
+
+EXAMPLES = r'''
+- name: Gather list of portgroup by VMs
+  vmware.vmware.portgroup_info:
+    hostname: "https://vcenter"
+    username: "username"
+    password: "password"
+    vm_names:
+      - vm-test1
+      - vm-test2
+'''
+
+RETURN = r'''
+vm_portgroup_info:
+    description:
+        - Dictionary of the requested VMs with the portgroup information
+    returned: On success
+    type: dict
+    sample: {
+        "vm1": [
+        {
+            "name": "Network Name",
+            "nic_mac_address": "00:00:00:00:00:00",
+            "nic_mac_type": "ASSIGNED",
+            "nic_type": "VMXNET3",
+            "port_id": "network-port-id",
+            "type": "STANDARD_PORTGROUP",
+            "vlan_id": "0",
+            "vswitch_name": "vSwitch0"
+        }]
+    }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.vmware.vmware.plugins.module_utils.vmware import PyVmomi
+from ansible_collections.vmware.vmware.plugins.module_utils.vmware_rest_client import VmwareRestClient
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+
+def get_standard_portgroup_vlan_vswitch(portgroup, pg_name):
```

Comment:
Maybe we should move all of these `get_` methods into a module util like `vmware_network` or something? They seem like they might be useful in other modules too

Reply:
You're right 
I thought about it and I wasn't sure but I will do it